src/cpp/fastdds/xtypes/type_conversion/dyn_type_tree.ipp

```diff
@@ -0,0 +1,1116 @@
+            {
+                ret = bitset_to_str(node, kind_str);
+                break;
+            }
+            case TK_ENUM:
+            {
+                ret = enum_to_str(node, kind_str);
+                break;
+            }
+            case TK_STRUCTURE:
+            {
+                ret = struct_to_str(node, kind_str);
+                break;
+            }
+            case TK_UNION:
+            {
+                ret = union_to_str(node, kind_str);
+                break;
+            }
+            default:
+            {
+                continue;
+            }
+        }
+
+        if (ret != RETCODE_OK)
+        {
+            return ret;
+        }
+
+        dyn_type_str += kind_str + "\n";
+        types_written.insert(node.info.type_kind_name);
+    }
+
+    // Write struct parent node at last, after all its dependencies
+    // NOTE: not a requirement for Foxglove IDL Parser, dependencies can be placed after parent
+    std::string struct_str;
+    ret = struct_to_str(root, struct_str);
+
+    if (ret != RETCODE_OK)
+    {
+        return ret;
+    }
+
+    dyn_type_str += struct_str;
+
+    return ret;
+}
+
+ReturnCode_t alias_to_str(
+        const utilities::collections::TreeNode<TreeNodeType>& node,
+        std::string& alias_str) noexcept
+{
+    if (node.info.dynamic_type->get_kind() != TK_ALIAS)
+    {
+        EPROSIMA_LOG_ERROR(DYN_TYPES, "Type is not an alias.");
+        return RETCODE_BAD_PARAMETER;
+    }
+
+    ReturnCode_t ret = RETCODE_OK;
+
+    TypeDescriptor::_ref_type type_descriptor {traits<TypeDescriptor>::make_shared()};
+    ret = node.info.dynamic_type->get_descriptor(type_descriptor);
+
+    if (ret != RETCODE_OK)
+    {
+        return ret;
+    }
+
+    // Find the base type of the alias
+    std::string base_type_kind_str;
+    ret = type_kind_to_str(type_descriptor->base_type(), base_type_kind_str);
+
+    if (ret != RETCODE_OK)
+    {
+        return ret;
+    }
+
+    alias_str = "typedef ";
+    alias_str += base_type_kind_str + " ";
+    alias_str += type_descriptor->name().to_string() + ";\n";
+
+    return ret;
+}
+
+ReturnCode_t bitmask_to_str(
+        const utilities::collections::TreeNode<TreeNodeType>& node,
+        std::string& bitset_str) noexcept
+{
+    if (node.info.dynamic_type->get_kind() != TK_BITMASK)
+    {
+        EPROSIMA_LOG_ERROR(DYN_TYPES, "Type is not a bitmask.");
+        return RETCODE_BAD_PARAMETER;
+    }
+
+    ReturnCode_t ret = RETCODE_OK;
+
+    TypeDescriptor::_ref_type type_descriptor {traits<TypeDescriptor>::make_shared()};
+    ret = node.info.dynamic_type->get_descriptor(type_descriptor);
+
+    if (ret != RETCODE_OK)
+    {
+        return ret;
+    }
+
+    if (type_descriptor->bound().size() != 1)
+    {
+        EPROSIMA_LOG_ERROR(DYN_TYPES, "Bitmask type must have exactly one bound.");
+        return RETCODE_BAD_PARAMETER;
+    }
+
+    // Annotation with the bitmask size
+    static constexpr std::uint32_t DEFAULT_BITMASK_SIZE = 32;
```

Comment:
Can't you access the default value instead of hardcoding it?

Reply:
The default value is not accessible.